src/index.js

```diff
@@ -102,6 +102,12 @@ export default function fetch(url, opts) {
 					requestOpts.headers.delete('content-length');
 				}
 
+				// for 307 redirects, the body of the original request is reused to perform the redirected request
+				// https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307
+				if (res.statusCode === 307) {
+					requestOpts.body = request.body;
```

Comment:
Two comments:

This step should be executed for all status codes other than 303. This spec-mandated behavior is echoed in Chrome.

We should reject the promise with an FetchError here if `request.body` is a readable stream. See https://fetch.spec.whatwg.org/#concept-http-redirect-fetch, step 9. ("*request*’s body’s source is null" is an oblique way of saying "*request* is backed by a stream.) It makes sense too not to consume the same stream multiple times.